src/server/scripts/EasternKingdoms/ZulGurub/boss_mandokir.cpp

```diff
@@ -251,31 +304,74 @@ class boss_mandokir : public CreatureScript
                             Talk(SAY_OHGAN_DEAD);
                         }
                         else
+                        {
                             events.ScheduleEvent(EVENT_CHECK_OHGAN, 1000);
+                        }
                         break;
                     case EVENT_WATCH_PLAYER:
                         if (Unit* player = SelectTarget(SelectTargetMethod::Random, 0, 100, true))
                         {
                             DoCast(player, SPELL_WATCH);
                             Talk(SAY_WATCH, player);
                         }
-                        events.ScheduleEvent(EVENT_WATCH_PLAYER, urand(12000, 15000));
+                        events.ScheduleEvent(EVENT_WATCH_PLAYER, urand(12000, 24000));
                         break;
                     case EVENT_CHARGE_PLAYER:
                         DoCast(SelectTarget(SelectTargetMethod::Random, 0, 40, true), SPELL_CHARGE);
-                        events.ScheduleEvent(EVENT_CHARGE_PLAYER, urand(22000, 30000));
+                        events.ScheduleEvent(EVENT_FRIGHTENING_SHOUT, 1500);
+                        if (Unit* mainTarget = SelectTarget(SelectTargetMethod::MaxThreat, 0, 100.0f))
+                        {
+                            me->GetThreatMgr().modifyThreatPercent(mainTarget, -100);
+                        }
+                        events.ScheduleEvent(EVENT_CHARGE_PLAYER, urand(30000, 40000));
+                        break;
+                    case EVENT_EXECUTE:
+                        if (me->GetVictim() && me->GetVictim()->HealthBelowPct(20))
+                        {
+                            DoCastVictim(SPELL_EXECUTE, true);
+                        }
+                        events.ScheduleEvent(EVENT_EXECUTE, urand(7000, 14000));
+                        break;
+                    case EVENT_FRIGHTENING_SHOUT:
+                        DoCastAOE(SPELL_FRIGHTENING_SHOUT);
                         break;
+                    case EVENT_CLEAVE:
+                        {
+                            std::list<Unit*> meleeRangeTargets;
+                            auto i = me->GetThreatMgr().getThreatList().begin();
+                            for (; i != me->GetThreatMgr().getThreatList().end(); ++i)
+                            {
+                                Unit* target = (*i)->getTarget();
+                                if (me->IsWithinMeleeRange(target))
+                                {
+                                    meleeRangeTargets.push_back(target);
+                                }
+                            }
+                            std::list<Unit*, std::allocator<Unit*>>::iterator itr;
+                            for (itr = meleeRangeTargets.begin(); itr != meleeRangeTargets.end(); ++itr)
```

Comment:
What's the purpose of this loop? It doesn't do anything with itr.

Reply:
forgot about it while working on other options, removed.